src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -695,6 +696,10 @@ ssize_t sockinfo_tcp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 	si_tcp_logfunc("tx: iov=%p niovs=%d dummy=%d", p_iov, sz_iov, is_dummy);
 	lock_tcp_con();
 
+	if (unlikely(MCE_SPEC_STAC == m_mce_spec)) {
```

Comment:
1. Please explain why calling rx_wait_helper() occurs inside lock_tcp_con() and not outside of it
2. I feel uncomfortable that we are checking m_mce_spec on the fast path for practically never-expected scenario.
Can you come with alternative suggestions?